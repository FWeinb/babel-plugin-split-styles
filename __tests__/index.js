import pluginTester from 'babel-plugin-tester';
import plugin from '../src';

pluginTester({
  plugin,
  babelOptions: {},
  tests: [
    {
      title: 'Only static css',
      code: '<div style={{color: "green"}} />',
      snapshot: true
    },
    {
      title: 'Dynamic styles should stay in `style`',
      code: '<div style={{color: a ? "red" : "green"}} /> ',
      snapshot: true
    },
    {
      title: 'Extract static from dynamic',
      code: '<div style={{background: "blue", color: a ? "red" : "green"}} />',
      snapshot: true
    },
    {
      title: 'Bail if `css` prop was used',
      code: '<div css={{background: "blue"}} style={{color: "green"}} />',
      snapshot: true
    }
  ]
});
