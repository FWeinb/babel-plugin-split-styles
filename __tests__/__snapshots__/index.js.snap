// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-split-styles Dynamic styles should stay in \`style\`: Dynamic styles should stay in \`style\` 1`] = `
"
<div style={{color: a ? \\"red\\" : \\"green\\"}} />

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

<div style={{
  color: a ? \\"red\\" : \\"green\\"
}} />;
"
`;

exports[`babel-plugin-split-styles Extract static from dynamic: Extract static from dynamic 1`] = `
"
<div style={{background: \\"blue\\", color: a ? \\"red\\" : \\"green\\"}} />

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

<div css={{
  background: \\"blue\\"
}} style={{
  color: a ? \\"red\\" : \\"green\\"
}} />;
"
`;

exports[`babel-plugin-split-styles Ignore bound jsx elements: Ignore bound jsx elements 1`] = `
"
const div = () => {}; <div style={{some: \\"other\\", stuff}} />

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

const div = () => {};

<div style={{
  some: \\"other\\",
  stuff
}} />;
"
`;

exports[`babel-plugin-split-styles Ignore non identifier as element name: Ignore non identifier as element name 1`] = `
"
<animated.div style={{some: \\"other\\", stuff}} />

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

<animated.div style={{
  some: \\"other\\",
  stuff
}} />;
"
`;

exports[`babel-plugin-split-styles Only static css: Only static css 1`] = `
"
<div style={{color: \\"green\\"}} />

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

<div css={{
  color: \\"green\\"
}} />;
"
`;

exports[`babel-plugin-split-styles Should bail if \`css\` property exists: Should bail if \`css\` property exists 1`] = `
"
<div css={{}} style={{background: \\"blue\\"}} />

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

<div css={{}} style={{
  background: \\"blue\\"
}} />;
"
`;

exports[`babel-plugin-split-styles Should bail on non object expression: Should bail on non object expression 1`] = `
"
<div style={a} />

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

<div style={a} />;
"
`;

exports[`babel-plugin-split-styles Should work with nested object: Should work with nested object 1`] = `
"
<div style={{color: \\"green\\", \\"&\\": { \\":hover\\": { color: \\"blue\\"}}}} />

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

<div css={{
  color: \\"green\\",
  \\"&\\": {
    \\":hover\\": {
      color: \\"blue\\"
    }
  }
}} />;
"
`;
